### PR TITLE
ExecutionModule: Mark state texture IDs as client-sided

### PR DESCRIPTION
**Pull request by Sturmlilie**
Fixes crash on dedicated server.

`Git bisect saves lives.`  
(Also could you make a new CurseForge release while at it? :>)

### CODE REVIEW

File: src/main/java/li/cil/tis3d/common/module/ExecutionModule.java [hunk 1]
```diff
@@ -64,12 +64,15 @@
...
     }
 
-    private static final Identifier[] STATE_LOCATIONS = new Identifier[]{
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_IDLE,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_ERROR,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_RUNNING,
-        Textures.LOCATION_OVERLAY_MODULE_EXECUTION_WAITING
-    };
+    @Environment(EnvType.CLIENT)
+    private static final class RenderData {
```
Comment (fnuecke) [inline src/main/java/li/cil/tis3d/common/module/ExecutionModule.java:68]: Feel free to only answer this only here or on the 1.16 pr comment: does adding the client-only annotation directly to the field not world? It appears to work in the `DisplayModule`; does this being a static field make a difference?

Comment (Sturmlilie) [reply src/main/java/li/cil/tis3d/common/module/ExecutionModule.java:68]: This was a long time ago, but I remember trying a lot of ways and settling on the one which had the least amount of changes. I'll try again.

**Comment by Sturmlilie** [comment]
So, this crashes:
```java
@Environment(EnvType.CLIENT)
static final Identifier[] STATE_LOCATIONS = new Identifier[]{
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_IDLE,
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_ERROR,
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_RUNNING,
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_WAITING
};
```
and this
```java
static final Identifier[] STATE_LOCATIONS = new Identifier[]{
    @Environment(EnvType.CLIENT)
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_IDLE,
    @Environment(EnvType.CLIENT)
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_ERROR,
    @Environment(EnvType.CLIENT)
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_RUNNING,
    @Environment(EnvType.CLIENT)
    Textures.LOCATION_OVERLAY_MODULE_EXECUTION_WAITING
};
```
doesn't compile. Regarding the DisplayModule: That was not an array type variable. So I'm kinda out of options, if you know another way I'd be all ear :shrug:

**Comment by fnuecke** [comment]
Right you are. [The latest docs even make this clear](https://github.com/FabricMC/fabric-loader/blob/fbca3acf6276cbccd6e13dff09a0112bd40c7fe2/src/main/java/net/fabricmc/api/Environment.java#L30) (the ones my 1.15 dev env downloads doesn't, hence my expectations). Thanks for checking!

In that case I'd agree this is the most readable solution. A lambda that checks `net.fabricmc.loader.api.FabricLoader.getInstance().getEnvironmentType()` looks pretty noisy.

**Pull request merged**